src/component.jl

```diff
@@ -125,96 +115,40 @@ function assign_new_uuid_internal!(component::InfrastructureSystemsComponent)
     return
 end
 
-"""
-Attach an attribute to a component.
-"""
-function attach_supplemental_attribute!(
-    component::InfrastructureSystemsComponent,
-    attribute::T,
-) where {T <: SupplementalAttribute}
-    attribute_container = get_supplemental_attributes_container(component)
-
-    if !haskey(attribute_container, T)
-        attribute_container[T] = Dict{Base.UUID, T}()
-    end
-
-    uuid = get_uuid(attribute)
-    if haskey(attribute_container[T], uuid)
-        throw(
-            ArgumentError(
-                "Supplemental attribute $uuid is already attached to $(summary(component))",
-            ),
-        )
-    end
-    attribute_container[T][uuid] = attribute
-    @debug "SupplementalAttribute type $T with UUID $uuid) stored in component $(summary(component))" _group =
-        LOG_GROUP_SYSTEM
-    return
-end
-
 """
 Return true if the component has supplemental attributes of the given type.
 """
 function has_supplemental_attributes(
```

Comment:
Why change in the order of operations.

Reply:
I think I did that for consistency with `has_time_series`. I really dislike this ordering question and wish we had gone our own way. I'm not opposed to changing it. However, my thought is that it is more intuitive to put the component before the type, similar to `haskey(Dict, my_key)`.
```
"""
Return true if the component or supplemental attribute has time series data of type T.
"""
function has_time_series(
    val::TimeSeriesOwners,
    ::Type{T},
) where {T <: TimeSeriesData}
```

Reply:
Both now exist.